File: sources/Android/android-communications/library/src/sdk/java/com/polar/sdk/api/model/PolarOfflineRecordingEntry.kt
// Copyright © 2022 Polar Electro Oy. All rights reserved.
package com.polar.sdk.api.model

import com.polar.sdk.api.PolarBleApi
import java.util.*

/**
 * Polar offline recording entry container.
 */
data class PolarOfflineRecordingEntry(
    /**
     * Recording entry path in device.
     */
    @JvmField
    val path: String,
    /**
     * Recording size in bytes.
     */
    @JvmField
    val size: Long,
    /**
     * The date and time of the recording entry i.e. the moment recording is started
     */
    @JvmField
    val date: Date,
    /**
     * data type of the recording
     */
    @JvmField
    val type: PolarBleApi.PolarDeviceDataType
)

File: sources/iOS/ios-communications/Sources/PolarBleSdk/sdk/api/model/PolarOfflineRecordingEntry.swift
//  Copyright © 2022 Polar. All rights reserved.

import Foundation

/// Polar offline recording entry container.
public struct PolarOfflineRecordingEntry : Equatable {
    /// Recording entry path in device.
    public let path: String
    /// Recording size in bytes.
    public let size: UInt
    /// The date and time of the recording entry i.e. the moment recording is started
    public let date: Date
    ///  data type of the recording
    public let type: PolarDeviceDataType
    
    public init(path: String, size: UInt, date: Date, type: PolarDeviceDataType) {
        self.path = path
        self.size = size
        self.date = date
        self.type = type
    }
}
